### DISCUSSION TITLE
header does not match what was computed. (Service: Amazon S3; Status Code: 400; Error Code: XAmzContentSHA256Mismatch;

### DISCUSSION THREAD

**Discussion by liuhao0826** [aws/aws-sdk-java#3155] [Q&A]
Using the setObjectAcl method to set the file to public read, an exception occurred in the service deployed on one server, but there was no problem during my local and OSS testing. Could you please provide some solutions?
![1729741585058](https://github.com/user-attachments/assets/77f92724-abb0-463f-b0c1-c9ed1c13f4cf)

**Comment by liuhao0826** [comment]
and i try getObject method ,   it is also this same error only in the prod server ，no error in my local

**Reply by debora-ito** [reply]
The error code and the request ID are not in the usual s3 format (e.g. error code XAmzContentSHA256Mismatch). If the prod environment is using a third-party endpoint, the issue is most likely on the third-party side.
